### ISSUE TITLE
Improve evolution column in all websites tab

### ISSUE THREAD

**Issue by fdellwing** [matomo-org/matomo#13124]
So, there is something going on my nerves for quite some time now:

If you had 0 visits yesterday and any visit today, it shows 100%.
If you had 1 visit yesterday and more than 2 today it shows a value more than 100%.

This is mathematical correct, but in imho a big problem.

Take the following scenario:
Changes are like 0 -> 500 and 1 -> 20. Than it shows 100% for the first and 1900% for the second evolution. **Implying that the second one increased more than the first one, which is absolutely wrong.**

I do not really have a solution right now, but maybe someone else? Or do you not see a problem with that?

![matomo1](https://user-images.githubusercontent.com/22813377/42158707-85a25bd2-7df1-11e8-9801-1d6fa974a6e5.png)

**Comment by tsteur** [comment]
Yeah I know exactly what you mean. Not sure how to fix though, maybe in the UI could show something like "New" but I presume the API would maybe still expect some number or so.

**Comment by fdellwing** [comment]
I thought about using `&infin;` (&infin;) instead of 100%, but that is also not a valid number, so duh.

**Comment by fdellwing** [comment]
I thought about this some more.

Instead of changing the output, it might be better to change the output?

https://github.com/matomo-org/matomo/blob/c6eb52339f3fc496ac43b55c4a0de8d161edbff4/plugins/CoreHome/Columns/Metrics/EvolutionMetric.php#L109-L111

Here it is hard coded to return 100% if the old value (divisor) is 0. Well we could instead provide an arbitrary divisor for the calculation.

I think there are 2 possible arbitrary numbers to choose from:

`return Piwik::getQuotientSafe($dividend, 1, $this->quotientPrecision + 2);`

Will return the same as if the old value was 1.

`return Piwik::getQuotientSafe($dividend, 1.0E-10, $this->quotientPrecision + 2);`

Will return a huge number and is easily distinguishable from the old value 1.

What are your thoughts with this? @tsteur

**Comment by tsteur** [comment]
Not really sure what is best here. Maybe @mattab has some thoughts.

**Comment by dev-101** [comment]
This might help:
https://stackoverflow.com/questions/19908431/how-to-calculate-percentage-when-old-value-is-zero

**Comment by dev-101** [comment]
Anyway, once I spoke with someone who was in financial business / calculations and told me that if initial value is 0, than whatever next positive value comes up (e.g. next month) it's assumed to be 100%. Because, strictly mathematically speaking, it would be +inf (division by 0). I think it is just an assumption / by convention, depending on department, field etc. By that logic, if you dropped from a positive value to 0, it should be -100% by that very same convention. And, if you think about it, then it makes sense.
